### PR TITLE
#176 Add Platform theme

### PR DESCRIPTION
**Pull request by aqwert**
PlatformTheme allows for under simple use cases a basic common theme with color. If any substantial differences between the material and cupertino themes are needed then setting the `materialTheme` and `cupertinoTheme` properties will be better

Adding documentation update in another commit

### CODE REVIEW

File: lib/src/platform_theme.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    this.accentColor,
+    this.bottomAppBarColor,
+    this.scaffoldBackgroundColor,
+    this.materialTheme,
+    this.cupertinoTheme,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Theme(
```
Comment (aqwert) [inline lib/src/platform_theme.dart:42]: May be better to extend InheritedTheme so changes can update

File: lib/src/platform_theme.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      primaryColor: primaryColor ?? theme.primaryColor,
+      accentColor: accentColor ?? theme.accentColor,
+      scaffoldBackgroundColor:
+          scaffoldBackgroundColor ?? theme.scaffoldBackgroundColor,
+      bottomAppBarColor: bottomAppBarColor ?? theme.bottomAppBarColor,
+    );
+  }
+
+  CupertinoThemeData _cupertinoTheme() {
+    final theme = cupertinoTheme ?? const CupertinoThemeData();
```
Comment (aqwert) [inline lib/src/platform_theme.dart:65]:
```suggestion
    final theme = cupertinoTheme ??  MaterialBasedCupertinoThemeData(
          materialTheme: materialData,
        )
```

**Comment by fotiDim** [comment]
Could we have light and dark theme support in `PlatformTheme`? Right now I have resorted to:
```
PlatformApp(
...
             cupertino: (_, __) => CupertinoAppData(
              theme: MaterialBasedCupertinoThemeData(
                  materialTheme:
                      isDark ? materialDarkTheme : materialLightTheme),
            ),
...
)
```
which works but it would be nice if it was abstracted away in `PlatformTheme`.

**Comment by aqwert** [comment]
@fotiDim I am still working on how light and dark theme will work. `CupertinoApp` does not have a dark mode property, instead it uses `CupertinoDynamicColor`

**Pull request closed without merging**